beacon-chain/cache/payload_id.go

```diff
@@ -65,9 +66,16 @@ func (f *ProposerPayloadIDsCache) PrunePayloadIDs(slot types.Slot) {
 	f.Lock()
 	defer f.Unlock()
 
-	for s := range f.slotToProposerAndPayloadIDs {
+	for k := range f.slotToProposerAndPayloadIDs {
+		s := types.Slot(bytesutil.BytesToUint64BigEndian(k[:8]))
 		if slot > s {
-			delete(f.slotToProposerAndPayloadIDs, s)
+			delete(f.slotToProposerAndPayloadIDs, k)
 		}
 	}
 }
+
+func idKey(slot types.Slot, r [32]byte) [32]byte {
+	var k [32]byte
+	copy(k[:], append(bytesutil.Uint64ToBytesBigEndian(uint64(slot)), r[:]...))
```

Comment:
this basically just copies the uint64 bytes + the first 24-bytes of the root ? Why not just make the key 40 bytes and have the whole root accounted for in it.